### DISCUSSION TITLE
how to config appium uiautomator2 not to wait for full content?

### DISCUSSION THREAD

**Discussion by genopapashvili** [appium/appium#16469] [General]
Some pages that are active (for example there is clock or timer on the screen) need much time to take out the XML, sometimes it does not take out any result. I wonder if there is any way to disable full XML document loading?

**Comment by jlipps** [comment]
There is no way to disable XML document loading, other than simply not calling page source or using the xpath strategy.

**Reply by genopapashvili** [reply]
Thank you for your answer 

so it happens because of xpath?
then I am able to parse full document in my side but can I create an element instance by the xml document? 
or if there is a way to get element by the coordinate FE X Y? 
 

The product provider creates  app in react native and I am only able to get the elements by xpath I have no other way


Regards

**Reply by jlipps** [reply]
For XPath to work it must generate a source XML document, yes. There are definitely other ways to find elements than XPath. React Native produces valid native elements. You can find them by accessibility ID or uiautomator as well, which would not generate XML.
